custom_components/ha-google-home/api.py

```diff
@@ -86,28 +86,42 @@ def get_alarms_and_timers_from(self, device, endpoint):
     def get_google_devices_information(self):
         """Retrieves devices from glocaltokens"""
         _LOGGER.debug("Fetching data...")
+        offline_devices = []
         devices = self._client.get_google_devices()
 
         for device in devices:
             # To avoid key error's
             timers = []
             alarms = []
-
-            result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
-            if result:
-                timers = result.get(TIMERS)
-                alarms = result.get(ALARMS)
-
-                if not timers and not alarms:
-                    _LOGGER.error(
-                        "For device {device} - {error}".format(
-                            device=device.device_name, error=API_RETURNED_UNKNOWN
+            if device.ip:
+                result_json = self.get_alarms_and_timers_from(
+                    device, API_ENDPOINT_ALARMS
+                )
+                if result_json:
+                    timers = result_json.get(TIMERS)
+                    alarms = result_json.get(ALARMS)
+
+                    if not timers and not alarms:
+                        _LOGGER.error(
+                            "For device {device} - {error}".format(
+                                device=device.device_name, error=API_RETURNED_UNKNOWN
+                            )
                         )
-                    )
+            else:
+                offline_devices.append(device.device_name)
```

Comment:
You call the list `offline_devices` but only append their names. You can just append `device` object